mdw-workflow/assets/com/centurylink/mdw/base/StuckActivities.java

```diff
@@ -0,0 +1,172 @@
+import com.centurylink.mdw.common.service.ServiceException;
+import com.centurylink.mdw.config.PropertyException;
+import com.centurylink.mdw.config.PropertyManager;
+import com.centurylink.mdw.constant.PropertyNames;
+import com.centurylink.mdw.dataaccess.DatabaseAccess;
+import com.centurylink.mdw.model.user.UserAction;
+import com.centurylink.mdw.model.workflow.ActivityInstance;
+import com.centurylink.mdw.model.workflow.Process;
+import com.centurylink.mdw.model.workflow.WorkStatus;
+import com.centurylink.mdw.service.data.process.EngineDataAccessDB;
+import com.centurylink.mdw.service.data.process.ProcessCache;
+import com.centurylink.mdw.services.ServiceLocator;
+import com.centurylink.mdw.services.WorkflowServices;
+import com.centurylink.mdw.services.workflow.RoundRobinScheduledJob;
+import com.centurylink.mdw.util.CallURL;
+import com.centurylink.mdw.util.log.LoggerUtil;
+import com.centurylink.mdw.util.log.StandardLogger;
+
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+
+
+/**
+ * This script fixes stuck activities by failing them and then retrying them
+ * A stuck activity means it's "In Progress" status and older than {ActivityAgeInMinutes}
+ * Add following to mdw.yaml
+timer.task:
+ StuckActivities:
+    TimerClass: com.centurylink.mdw.timer.cleanup.StuckActivities
+    Schedule: 30 2 * * *    # to run hourly at 30 minutes past the hour use : Schedule: 30 * * * ? *
+    ActivityAgeInMinutes: 1800 # How old activity instance should be to be a candidate for fixing
+    MaximumActivities: 10  # How many stuck activity instances to be retried in each run
+
+ * if you need to make change in above properties then first delete the db entry by identifying the row using
+ * this sql:  select * from event_instance where event_name like '%ScheduledJob%'
+ * Then re-start the server/instance for new clean-up properties to be effective.
+ *
+ * The query used to identify stuck activities will perform a full table scan, so
+ * consider creating an index on ACTIVITY_INSTANCE table if it is causing performance issues.
+ */
+//scheduledjob annotation is added and this is moved to the base package . values are read from the property file.
+@ScheduledJob(value="StuckActivities", schedule="${props['mdw.StuckActivities.job.scheduler']}", enabled="${props['mdw.StuckActivities.job.enabled']}", defaultEnabled= false, isExclusive=true)
+public class StuckActivities extends RoundRobinScheduledJob implements com.centurylink.mdw.model.monitor.ScheduledJob  {
+
+    private static boolean running = false;
+    private static StandardLogger logger = LoggerUtil.getStandardLogger();
+
+    @Override
+    public void run(CallURL args) {
+        if (!running) {
+            running = true;
+            logger.info("methodEntry-->StuckActivities.run()");
+            int activityAge;  //Seconds
+            int maxActivities;
+
+                try {
```

Comment:
You are still reading old timer task properties.  Instead you should read the new properties as described in your documentation updates.